components/frontend-react/src/sagas/downloadSaga.js

```diff
@@ -0,0 +1,49 @@
+import { take, takeEvery, put, call } from 'redux-saga/effects';
+import { startAllDownloads, killAllDownloads, addNewDownload, fetchDownloads, deleteDownload } from '../API/Downloads';
+import { downloadActions } from '../constants';
+
+function* startAllDownloadsWorkerSaga () {
+  yield call(startAllDownloads);
```

Comment:
Sagas should handle it's effects. Use try catch . What if your requests fails?  @agentmilindu WDYT?

Reply:
Agree with you. The failure scenario also should be considered.